### DISCUSSION TITLE
Can i show relation field only when i fetch specific item?

### DISCUSSION THREAD

**Discussion by chotirat-c** [strapi/strapi#8176] [Legacy Topics]
just like i have http://localhost:1337/restaurants which return list of every restaurants. but restaurants has many reviews.
so when i fetch from /restaurants  it return list of reviews also which is too much for server i think. so is there any way to return list of review only when i fetch from specific restaurant like /restaurants/:id ?. thank you so much

**Comment by derrickmehaffy** [comment]
Hi @mikasaloli 

We are planning to remove the GitHub discussions in early 2021, in the future can you please ask your questions on our [Forum](https://forum.strapi.io/). As your question is easy to answer, I'll go ahead and do so.

Yes you can customize the populate for the `find` service and leave the `findOne` as the default.

You will need to take the default `find` service from our documentation: https://strapi.io/documentation/v3.x/concepts/services.html#find

You will place this in the `./api/restaurant/services/restaurant.js` and you will want to manually define that `population` variable with an array.

```js
module.exports = {
  /**
   * Promise to fetch all records
   *
   * @return {Promise}
   */
  find(params, populate) {
    return strapi.query('restaurant').find(params, []);
  },
};
```

Depending on if you have other relations defined, you will need to add those relation fields into that array like say `[ "owner" ]` (just an example)

**Reply by asdutoit** [reply]
Hi @derrickmehaffy , 

I have a similar question and have implemented the code as per your proposed solution but no success. 

Sample of my code: 
```Javascript
module.exports = {
  find(params, populate) {
    return strapi
      .query("onts")
      .find(params, ["address_unit", "address_unit.address_complex"]);
  },
};
```

And here is the output when I run the API fetch:

```
[
  {
    "model": "PMG1005",
    "activity_histories": [],
    "images": [],
    "_id": "5f6da2849e88f2001791778b",
    "make": "Zyxel",
    "serial_number": "5A594F43806616F8",
    "createdAt": "2020-09-25T07:55:48.236Z",
    "updatedAt": "2020-10-13T14:11:27.620Z",
    "__v": 0,
    "address_unit":     //  <------------
      "_id": "5f6da2969e88f2001791778c",
      "street": "Edgemead dr ",
      "unit_no": "106",
      "latitude": -33.869214,
      "longitude": 18.542392,
      "createdAt": "2020-09-25T07:56:06.733Z",
      "updatedAt": "2020-10-13T14:11:41.075Z",
      "__v": 0,
      "address_complex": "5f1435a58d43c6619be1ae42",   //  <----   EXPECTED THIS TO RETURN THE RECORD (MONGODB DOCUMENT)
      "ont": "5f6da2849e88f2001791778b",
      "resident": "5f6da2eb9e88f2001791778d",
      "block_name": null,
      "id": "5f6da2969e88f2001791778c"
    },
    "id": "5f6da2849e88f2001791778b"
  },
...,
...
]

Could you possibly identify the error in my code?

Appreciate the help!

**Reply by asdutoit** [reply]
Nevermind, I came right.  Thanks
